ocaml/xapi/xapi_pvs_proxy.ml

```diff
@@ -32,9 +32,18 @@ let create ~__context ~site ~vIF =
       ) ;
   Helpers.assert_is_valid_ref ~__context ~name:"site" ~ref:site ;
   Helpers.assert_is_valid_ref ~__context ~name:"VIF" ~ref:vIF ;
-  let device = Db.VIF.get_device ~__context ~self:vIF in
-  if device <> "0" then
-    raise Api_errors.(Server_error (invalid_device, [device])) ;
+  let device = Db.VIF.get_device ~__context ~self:vIF |> int_of_string in
+  let min_device =
+    let open Xapi_database.Db_filter_types in
+    let vm = Db.VIF.get_VM ~__context ~self:vIF in
+    Db.VIF.get_records_where ~__context
+      ~expr:(Eq (Field "VM", Literal (Ref.string_of vm)))
+    |> List.fold_left
```

Comment:
It might be clearer to define a function `minimum` that takes a non-empty list of integers.

Reply:
```
val minimum: min:int -> int list -> int (** minimum of the list or min if the list is empty *)
...
client code like

| hd::tl -> minimum min:hd tl
..
```
Maybe it's nice but verbose.

Reply:
IMO the current fold_left is more to the point.